### PR TITLE
Alireza/none/hotfix bugs

### PR DESCRIPTION
**Pull request by alimosaed**
### What is the purpose of this change?
There are some package dependencies. Also, it fails to log Signal interrupts properly when the broker is attempting to reconnect.

### How is this accomplished?
Add dependencies to the .toml file and handled interrupts.

### Anything reviews should focus on/be aware of?

### CODE REVIEW

File: src/solace_ai_connector/common/messaging/solace_messaging.py [hunk 1]
```diff
@@ -136,8 +140,16 @@ def connect(self):
...
 
-        # Blocking connect thread
-        self.messaging_service.connect()
+        try:
+            # Blocking connect thread
+            self.messaging_service.connect()
+        except KeyboardInterrupt:
+            log.error("Interrupt received. Disconnecting from messaging service.")
+            self.disconnect()
+            exit()
```
Comment (cyrus2281) [inline src/solace_ai_connector/common/messaging/solace_messaging.py:149]: You shouldn't exit the program from inside a component, there are still a lot other components that need to be cleaned up

**Review by cyrus2281 — CHANGES_REQUESTED** [review]
Check comment

File: src/solace_ai_connector/common/messaging/solace_messaging.py (hunk 1 above)
Comment (cyrus2281) [reply src/solace_ai_connector/common/messaging/solace_messaging.py:149]: Interrupt signal should not be caught from inside a component, the logic for that is here https://github.com/SolaceDev/solace-ai-connector/blob/main/src/solace_ai_connector/solace_ai_connector.py#L99

Comment (alimosaed) [reply src/solace_ai_connector/common/messaging/solace_messaging.py:149]: @cyrus2281 I moved the exception handler around the `app.run` in the` main.py`. 
Note that this error happens when the workflow is initialized by the `run` method.

**Review by cyrus2281 — APPROVED** [review]
Looks great, thanks

**Pull request merged**